File: Lab2.3/C++/StringLine.cpp
#include "StringLine.h"

StringLine::StringLine(const string& text) : text(text) {}

string StringLine::getText() const {
    return text;
}

string StringLine::getKeyFromFirstLetters() const {
    if (text.empty()) return "";

    istringstream iss(text);
    string word;
    string result;

    while (iss >> word) {
        if (!word.empty()) {
            result += word[0];
        }
    }
    return result;
}

File: Lab2.3/С#/StringLine.cs
public class StringLine
{
    private string text;

    public StringLine(string text)
    {
        this.text = text;
    }

    public string Text => text;

    public string GetKeyFromFirstLetters()
    {
        return string.Join("",
            text.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries)
                .Select(word => word.Length > 0 ? word[0].ToString() : ""));
    }

    public void CapitalizeFirstLetters()
    {
        text = string.Join(" ",
            text.Split(' ')
                .Select(word => word.Length > 0
                    ? char.ToUpper(word[0]) + word.Substring(1).ToLower()
                    : ""));
    }
}